epuck_ros2_cpp/include/epuck_ros2_cpp/mpu9250.hpp

```diff
@@ -0,0 +1,50 @@
+// Copyright 1996-2020 Cyberbotics Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef EPUCK_ROS2_CPP__MPU9250_HPP_
+#define EPUCK_ROS2_CPP__MPU9250_HPP_
+
+#include <inttypes.h>
+#include <memory>
+#include <vector>
+
+#include "epuck_ros2_cpp/i2c_wrapper.hpp"
+
+class MPU9250 {
+public:
+  MPU9250() = delete;
+  explicit MPU9250(std::shared_ptr<I2CWrapper> i2c);
+  void calibrate();
+  void read();
+
+  std::vector<float> getAngularVelocity();
+  std::vector<float> getLinearAcceleration();
+
+private:
+  int read_register(char reg, char *data, int size);
+  void read_raw(int16_t *rawAccelerometer, int16_t *rawGyroscope);
+
+  char mBuffer[6];
+  float mGyroscope[3];
+  float mAccelerometer[3];
+  int16_t mRawGyroscope[3];
+  int16_t mRawAccelerometer[3];
```

Comment:
As these raw value are only used in the `MPU9250::read` they might be declared as local variables of the function instead.

Reply:
Same for `mBuffer` actually.

Reply:
I didn't want to put them on the stack on every call. Maybe I am doing micro optimisations and it compiler would do it for me anyway?

Reply:
This looks indeed overkill to me as a few int16_t should not arm too much (and as you said this is maybe done by the compiler) for me keeping the class definition as simpler as possible is more important, and if you want to avoid putting them on the stack you can define them as static  to the function.

Reply:
Good idea! Moved as static

Reply:
It is good to put these variables inside the function scope, but I am not fully convinced they should be declared as static. This looks like a premature optimization hack. It might be that the compiler will store these values in some CPU registers (instead of the stack) and thus, it may turn out to outperform the static implementation. I would implement this kind of optimization hack only:
- after a bottleneck was clearly identified at this place.
- after making some performance benchmarks that demonstrate the `static` implementation is actually faster than the non-static one (or referring to an article that clearly demonstrates this).
- after adding some comments in the code that justify why the variables are declared as static (citing the article).

I have no problem if you want to leave the code like that, but I would request that at least you add a comment that explains that these variables are declared as static for performance reasons (ideally, citing an article demonstrating this).